algo/src/test/java/com/hartwig/actin/algo/evaluation/general/HasMaximumBMITest.java

```diff
@@ -0,0 +1,67 @@
+package com.hartwig.actin.algo.evaluation.general;
+
+import com.hartwig.actin.algo.datamodel.Evaluation;
+import com.hartwig.actin.algo.datamodel.EvaluationResult;
+import com.hartwig.actin.clinical.datamodel.ImmutableBodyWeight;
+import org.junit.Test;
+
+import java.time.LocalDate;
+import java.util.Arrays;
+import java.util.Collections;
+
+import static com.hartwig.actin.algo.evaluation.EvaluationAssert.assertEvaluation;
+import static org.junit.Assert.assertTrue;
+
+public class HasMaximumBMITest {
+    private final HasMaximumBMI function = new HasMaximumBMI(40);
+    private final LocalDate now = LocalDate.now();
+    private final LocalDate lastYear = now.minusYears(1);
+
+    @Test
+    public void shouldBeUndeterminedWhenNoBodyWeightsProvided() {
+        assertEvaluation(EvaluationResult.UNDETERMINED,
+                function.evaluate(GeneralTestFactory.withBodyWeights(Collections.emptyList())));
+    }
+
+    @Test
+    public void shouldBeUndeterminedWhenNoBodyWeightsProvidedWithExpectedUnit() {
+        assertEvaluation(EvaluationResult.UNDETERMINED, function.evaluate(GeneralTestFactory.withBodyWeights(
+                Collections.singleton(
+                        ImmutableBodyWeight.builder().date(now).value(70).unit("pound").build()
+                )
+        )));
+    }
+
+    @Test
+    public void shouldPassIfLatestWeightIsLessThanWarnThreshold() {
+        Evaluation evaluation = function.evaluate(GeneralTestFactory.withBodyWeights(Arrays.asList(
+                ImmutableBodyWeight.builder().date(now).value(70.57).unit("Kilogram").build(),
+                ImmutableBodyWeight.builder().date(lastYear).value(100.0).unit("Kilogram").build()
```

Comment:
Rather than referring to "Kilogram" should you not want to refer to HasMaximumBPI.EXPECTED_UNIT? I imagine you don't care what unit the expected unit is, as long as the weights are passed in the expected unit?

Reply:
I used different case here to test case-insensitive matching. We can change the tests to reference the constant if they become too brittle.

Reply:
Sounds good